src/macaron/slsa_analyzer/provenance/intoto/v1/__init__.py

```diff
@@ -11,9 +11,10 @@
 from macaron.slsa_analyzer.provenance.intoto.errors import ValidateInTotoPayloadError
 from macaron.util import JsonType
 
-# The full list of cryptographic algorithms supported in SLSA v1 provenance. These are used as keys within the digest
-#  set of the resource descriptors within the subject.
-# See: https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md
+# The full list of cryptographic algorithms supported in in-toto v0.1 and v1 provenance.
+# These are used as keys within the digest set of the resource descriptors within the subject.
+# For v1 see: https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md
+# For v0.1 see: https://github.com/in-toto/attestation/blob/main/spec/v0.1.0/field_types.md#DigestSet
 VALID_ALGORITHMS = [
```

Comment:
This might not be in scope of this PR.
I have a look at the definition for DigestSet in the v1 specification ([here](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md)) and I see three types:
1. [The standard crypto graphic hash algorithms](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#sha256-sha224-sha384-sha512-sha512_224-sha512_256-sha3_224-sha3_256-sha3_384-sha3_512-shake128-shake256-blake2b-blake2s-ripemd160-sm3-gost-sha1-md5) which is the ones that you put in this list (in https://github.com/oracle/macaron/issues/639)
2. [dirHash](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#dirhash)
3. [git related hash](https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#gitcommit-gittree-gitblob-gittag). This section contains the `gitCommit` value that you used for getting the commit hash of the source for SLSA provenance v1.0
I wonder if we want to support the second and third type as well :thinking:

Reply:
I think adding support for type three is in scope.